sdk/include/opentelemetry/sdk/trace/recordable.h

```diff
@@ -94,6 +120,19 @@ class Recordable
    * @param duration the duration to set
    */
   virtual void SetDuration(std::chrono::nanoseconds duration) noexcept = 0;
+
+private:
+  /**
+   * Maintain a static empty map that represents default attributes.
+   * Avoid constructing a new empty map everytime a call is made with default attributes.
+   */
+  static const nostd::shared_ptr<trace_api::KeyValueIterableView<std::map<std::string, int>>>
+      &GetDefaultAttributes() noexcept
```

Comment:
Let's call this `GetEmptyAttributes`, this is more accurate.

Reply:
Also, we should consider moving this as a function into a separate file and make it reusable for other components. I could see how we'd use something similar for in other instances, e. g. for metrics.

It might be placed into an additional header file in `include/opentelemetry/sdk/common`.

Reply:
Then we could also easily add unit tests for the function.

Reply:
I'll try that, thanks!